idl/join_node.idl.hh

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2023-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+namespace service {
+
+struct join_node_request_params {
+    raft::server_id host_id;
+    std::optional<raft::server_id> replaced_id;
+    std::vector<sstring> ignore_nodes;
+    sstring cluster_name;
+    sstring snitch_name;
+    sstring datacenter;
+    sstring rack;
+    sstring release_version;
+    uint32_t num_tokens;
+    uint32_t shard_count;
+    uint32_t ignore_msb;
+    std::vector<sstring> supported_features;
+};
+
+struct join_node_request_result {
+    struct ok {};
+    struct accepted {};
```

Comment:
What's the difference between ok and accepted? Maybe "ok" should be "already_a_member" to be clearer?